src/query/sql/src/planner/optimizer/rule/rewrite/rule_push_down_filter_join.rs

```diff
@@ -139,35 +137,13 @@ pub fn try_push_down_filter_join(
     let mut need_push = false;
 
     for predicate in predicates.into_iter() {
-        if is_falsy(&predicate) {
```

Comment:
I think we need `is_falsy` and `false_constant()`, if true, we can push `false_constant()` down, otherwise do nothing.